compute_modules/client/internal_query_client.py

```diff
@@ -220,9 +227,17 @@ def handle_job(self, job: Dict[str, Any]) -> None:
             self.logger.debug("Successfully executed job")
         except Exception as e:
             self.logger.error(f"Error executing job: {str(e)}")
-            result = self.get_failed_query(f"{str(e)}: {traceback.format_exc()}")
+            result = self.get_failed_query(e)
         self.logger.debug("Reporting result for job")
-        self.report_job_result(job_id, result)
+        if isinstance(result, Iterable) and not isinstance(result, dict):
```

Comment:
We need to check if it's an iterable instead of generator. Because user can stream result with a queue. I have an example. Will link tests in PR description.

Generator is also Iterable.

Dict is also Iterable so need to exclude it.

Reply:
replied below